test/integration/controllermanager/project/project/project_test.go

```diff
@@ -536,5 +537,250 @@ var _ = Describe("Project controller tests", func() {
+				updatedNamespacedCloudProfile := namespacedCloudProfile.DeepCopy()
+				updatedNamespacedCloudProfile.Spec.Kubernetes = &gardencorev1beta1.KubernetesSettings{
+					Versions: []gardencorev1beta1.ExpirableVersion{{Version: "1.25.1", ExpirationDate: futureExpirationDate}},
+				}
+				updatedNamespacedCloudProfile.Spec.MachineImages = []gardencorev1beta1.MachineImage{
+					{Name: "some-OS", Versions: []gardencorev1beta1.MachineImageVersion{
+						{ExpirableVersion: gardencorev1beta1.ExpirableVersion{Version: "1.1.1", ExpirationDate: futureExpirationDate}},
+					}},
+				}
+				Expect(testUserClient.Update(ctx, updatedNamespacedCloudProfile)).To(BeForbiddenError())
+			})
+
+			It("should allow gardener operators to modify a project's NamespacedCloudProfiles including special fields", func() {
+				By("Grant user operator-like permissions to modify special fields in NamespacedCloudProfile")
+				clusterRoleGardenerMember := &rbacv1.ClusterRole{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   "gardener.cloud:system:project-member",
+						Labels: map[string]string{"gardener.cloud/role": "project-member"},
+					},
+					Rules: []rbacv1.PolicyRule{
+						{
+							APIGroups: []string{"core.gardener.cloud"},
+							Resources: []string{"namespacedcloudprofiles"},
+							Verbs:     []string{"get", "list", "watch", "create", "patch", "update", "delete"},
+						},
+						{
+							APIGroups: []string{"core.gardener.cloud"},
+							Resources: []string{"namespacedcloudprofiles"},
+							Verbs:     []string{"modify-spec-kubernetes", "modify-spec-machineimages"},
+						},
+					},
+				}
+				Expect(testClient.Create(ctx, clusterRoleGardenerMember)).To(Succeed())
+				DeferCleanup(func() {
+					Expect(testClient.Delete(ctx, clusterRoleGardenerMember)).To(Succeed())
+				})
+
+				By("Add admin to project")
+				patch := client.MergeFrom(project.DeepCopy())
+				project.Spec.Members = append(project.Spec.Members, gardencorev1beta1.ProjectMember{
+					Subject: rbacv1.Subject{
+						APIGroup: rbacv1.GroupName,
+						Kind:     rbacv1.UserKind,
+						Name:     testUserName,
+					},
+					Role: "admin",
+				})
+				Expect(testClient.Patch(ctx, project, patch)).To(Succeed())
+
+				By("Ensure admin with proper role can update NamespacedCloudProfile.Spec.{Kubernetes,MachineImages}")
+				namespacedCloudProfile.Spec.Kubernetes = &gardencorev1beta1.KubernetesSettings{
+					Versions: []gardencorev1beta1.ExpirableVersion{{Version: "1.25.1", ExpirationDate: futureExpirationDate}},
+				}
+				namespacedCloudProfile.Spec.MachineImages = []gardencorev1beta1.MachineImage{
+					{Name: "some-OS", Versions: []gardencorev1beta1.MachineImageVersion{
+						{ExpirableVersion: gardencorev1beta1.ExpirableVersion{Version: "1.1.1", ExpirationDate: futureExpirationDate}},
+					}},
+				}
+				Eventually(func(g Gomega) {
+					g.Expect(testUserClient.Update(ctx, namespacedCloudProfile)).To(Succeed())
+				}).Should(Succeed())
```

Comment:
```suggestion
				Eventually(func() error {
					return testUserClient.Update(ctx, namespacedCloudProfile)
				}).Should(Succeed())
```

Nit, so you don't need double `Succeed` calls. Same for other occurrences.